mockgcp/mocksecretmanager/secrets.go

```diff
@@ -154,6 +161,10 @@ func (s *SecretsV1) UpdateSecret(ctx context.Context, req *pb.UpdateSecretReques
 			updated.Expiration = &pb.Secret_ExpireTime{
 				ExpireTime: req.Secret.GetExpireTime(),
 			}
+		case "ttl":
+			updated.Expiration = &pb.Secret_Ttl{
+				Ttl: req.Secret.GetTtl(),
```

Comment:
Nit: really?  I would have assumed it would set Expiration.ExpireTime, just like in create.  Still if this is what the logs tell us....

Reply:
Actually I don't know if we have test coverage of this in this PR (?).  It's not a big deal, because it's a mock (we can fix it when we need it), just that I would have guessed the other way!

In general I think the behaviour of this `ttl` field is very tricky.  Should we reconcile it (meaning secrets never expire)?  Should we only apply on create?  But if we do that, what happens when the user changes the TTL field in KRM?  What happens if they change a different field?

Reply:
Yes... here's what the SecertManager server tells us  https://screenshot.googleplex.com/9giJXnHPt3nMSwm.png

Reply:
Oh I see.  We set ttl, and then in `populateDefaultsForSecret` we convert it.  I guess that works, because then we can use the shared fieldmask update code at some point in the future!